driver/src/main/java/org/neo4j/driver/v1/Values.java

```diff
@@ -91,6 +94,10 @@ public static Value value( Object value )
     {
         if ( value == null ) { return NullValue.NULL; }
 
+        if ( value instanceof NodeValue || value instanceof RelationshipValue || value instanceof PathValue )
```

Comment:
I think it will be better to prohibit node, relationship and path in `PackStreamMessageFormatV1`. It is clear that struct packing code is the single point that writes structs to the channel. While `Values#value()` is just one of many factory methods.